src/EventStore.Client.Streams/EventStoreClient.Append.cs

```diff
@@ -92,105 +92,136 @@ public partial class EventStoreClient {
 		}
 
 		private async ValueTask<IWriteResult> AppendToStreamInternal(
-			CallInvoker callInvoker,
-			AppendReq header,
-			IEnumerable<EventData> eventData,
-			EventStoreClientOperationOptions operationOptions,
-			TimeSpan? deadline,
-			UserCredentials? userCredentials,
-			CancellationToken cancellationToken) {
-
-			using var call = new Streams.Streams.StreamsClient(
-				callInvoker).Append(EventStoreCallOptions.CreateNonStreaming(
-				Settings, deadline, userCredentials, cancellationToken));
+				CallInvoker callInvoker,
+				AppendReq header,
+				IEnumerable<EventData> eventData,
+				EventStoreClientOperationOptions operationOptions,
+				TimeSpan? deadline,
+				UserCredentials? userCredentials,
+				CancellationToken cancellationToken
+				) {
+			using var call = new Streams.Streams.StreamsClient(callInvoker).Append(
+					EventStoreCallOptions.CreateNonStreaming(Settings, deadline, userCredentials, cancellationToken)
+					);
 
 			IWriteResult writeResult;
+
 			try {
 				await call.RequestStream.WriteAsync(header).ConfigureAwait(false);
 
 				foreach (var e in eventData) {
-					_log.LogTrace("Appending event to stream - {streamName}@{eventId} {eventType}.",
-						header.Options.StreamIdentifier, e.EventId, e.Type);
-					await call.RequestStream.WriteAsync(new AppendReq {
-						ProposedMessage = new AppendReq.Types.ProposedMessage {
-							Id             = e.EventId.ToDto(),
-							Data           = ByteString.CopyFrom(e.Data.Span),
-							CustomMetadata = ByteString.CopyFrom(e.Metadata.Span),
-							Metadata = {
-								{Constants.Metadata.Type, e.Type},
-								{Constants.Metadata.ContentType, e.ContentType}
-							}
+					await call.RequestStream.WriteAsync(
+						new AppendReq {
+							ProposedMessage = new AppendReq.Types.ProposedMessage {
+								Id             = e.EventId.ToDto(),
+								Data           = ByteString.CopyFrom(e.Data.Span),
+								CustomMetadata = ByteString.CopyFrom(e.Metadata.Span),
+								Metadata = {
+									{ Constants.Metadata.Type, e.Type },
+									{ Constants.Metadata.ContentType, e.ContentType }
+								}
+							},
 						}
-					}).ConfigureAwait(false);
+					).ConfigureAwait(false);
 				}
-			} finally {
+
 				await call.RequestStream.CompleteAsync().ConfigureAwait(false);
+			} catch (RpcException) {
+				// Do nothing so that RpcExceptions propagate to the call.ResponseAsync and be translated by the TypedInterceptor
```

Comment:
How are the RPC exceptions propagated if we are catching them here? 
Asking cause its hard to understand the full scope of the execution just from this file.